### ISSUE TITLE
'linux' does not seem inclusive of 'unix' RID

### ISSUE THREAD

**Issue by cheenamalhotra** [dotnet/runtime#33046]
Hi Team,

I'm coming from issue dotnet/sqlclient#445 where providing `--runtime linux-x64` fails to load DLL from `runtimes\unix\...` folder.

RID ref: https://github.com/dotnet/runtime/blob/master/src/libraries/pkg/Microsoft.NETCore.Platforms/runtime.json

Is that expected behavior?

**Comment by janvorli** [comment]
cc: @ericstj

**Comment by ericstj** [comment]
Can you provide a repro that demonstrates what you claim is broken?

Here's how things are expected to work:
- If you publish with a runtime, then NuGet resolves the correct final assets for the application, writes this information in the project.assets.json, then the SDK copies out the correct set of things (determined by NuGet) to put in the published application's layout.  At runtime the host only has one DLL to choose from.
- If you publish without a runtime, then NuGet resolves all the possible runtime-targets, writes those to the project.assets.json, then the SDK uses this to generate a deps.json for the application and it copies all the possible runtime assets to the application.  At runtime the host reads the deps file and decides which RID-specific asset to load based on its current RID and the fallbacks listed in the shared framework deps file.

So if you can provide a simple repro, or the items listed here we can help you understand what's going wrong.

**Comment by cheenamalhotra** [comment]
Repro is available in original issue, copied below:

Clone this repo: https://github.com/davedx/sqlclient-repro

Then run:

`docker build -t efgs -f ./Dockerfile.custom.dotnet.3.1 .`
`docker run efgs`

**Comment by ericstj** [comment]
The bug here is that this docker file is publishing both a framework-dependent and a standalone app to the same directory.

https://github.com/davedx/sqlclient-repro/blob/755f9ac2a43c422b275015d28ba9633956e3d741/Dockerfile.custom.dotnet.3.1#L16-L19

dotnet build -c Release **-o /app**
dotnet publish -c Release **-o /app** --runtime linux-x64

The latter produces a self-contained app which is expecting to run without probing for runtimes, however the former already a bunch of files to the root directory which aren't runtime-specific.  

Here's the log from the publish:
```
Skipping target "_CopyResolvedFilesToPublishPreserveNewest" because all output files are up-to-date with respect to the input files.
```
So this skips copying *everything* because it's all the same or older than the files copied by the previous build.  To fix this use a clean directory for publishing with Runtime.

So this appears to be by-design since you're reusing the same directory.   If you disagree the correct place for a bug is https://github.com/dotnet/sdk

**Comment by davedx** [comment]
Hi @ericstj, original bug reporter here, thanks for your reply!

Now that I understand what is happening here it makes sense. It did cost us a quite some time to get to this point because previously, running dotnet build then dotnet publish behaved differently (publish pulled in the platform dll's) so this dockerfile worked properly for us for the last 2 years without any issues. I think, re-reading the documentation again [1][2] it would be good to have a warning about combining both build and publish like we were and what the consequences of this can be, because it's quite a big gotcha and it's difficult to diagnose what's going wrong.

"Skipping target "_CopyResolvedFilesToPublishPreserveNewest" because all output files are up-to-date with respect to the input files."

Thinking about it, I find the behaviour that it skips copying dependencies because it thinks they are already in the output directory kind of weird, and actually even incorrect? If it's supposed to copy the dependencies for the specified platform but does not verify that the correct platform specific versions of the dependencies are there, that seems like a bug to me... I think I will file a bug on the sdk repo.

[1] https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build
[2] https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish

**Comment by ericstj** [comment]
I tend to agree. Let’s ask the SDK to change behavior of publish to copy if different rather than newer.
